gsw/geostrophy.py

```diff
@@ -139,48 +141,43 @@ def distance(lon, lat, p=0):
         distance in meters between adjacent points.
 
     """
-    # This uses the algorithm from pycurrents rather than the one
-    # in GSW-Matlab.
+    lon, lat, = np.atleast_2d(lon), np.atleast_2d(lat)
 
-    if lon.ndim != 1 or lat.ndim != 1:
-        raise ValueError('lon, lat must be 1-D; found shapes %s and %s'
-                         % (lon.shape, lat.shape))
-    if lon.shape != lat.shape:
-        raise ValueError('lon, lat must have same 1-D shape; found %s and %s'
-                         % (lon.shape, lat.shape))
-    if p != 0:
-        if np.iterable(p) and p.shape != lon.shape:
-            raise ValueError('lon, non-scalar p must have same 1-D shape;'
-                             ' found %s and %s'
-                             % (lon.shape, lat.shape))
+    if (lon.size == 1) & (lat.size == 1):
+        raise ValueError('more than one point is needed to compute distance')
+    elif lon.ndim != lat.ndim:
+        raise ValueError('lon, lat must have the same dimension')
```

Comment:
This won't do in the long run; it doesn't tell the user what is wrong, and it is working with inputs *after* their shapes have been modified relative to what the user provided.  Leave it for now; I suspect it is just one of several places where the dimension handling can be changed.  It's messy because of the conflicts between trying to be consistent with Matlab (including the translated check functions), being internally consistent in Python, and the difference between the way the distance function is used compared to most other functions.